### PR TITLE
Fix map legend and colors

### PR DESCRIPTION
**Pull request by subinasr**
- Addresses #187 #150 

## Changes

* fix map legends and colors
* fix flag and country filter disappearing on reload

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

### CODE REVIEW

File: app/components/MapLabel/styles.css [hunk 1]
```diff
@@ -9,7 +9,8 @@
 
     .bar {
         border-radius: var(--dui-border-radius-card);
-        background: linear-gradient(90deg, rgba(215,48,39,1) 0%, rgba(255,255,191,1) 50%, rgba(26,152,80,1) 100%);
+        background: rgb(255,247,251);
+        background: linear-gradient(90deg, rgba(255,247,251,1) 0%, rgba(116,169,207,1) 50%, rgba(2,56,88,1) 100%);
```
Comment (tnagorra) [inline app/components/MapLabel/styles.css:13]: Why define two background?

File: app/views/Dashboard/Overview/MapView/index.tsx [hunk 2]
```diff
@@ -366,10 +352,13 @@ function MapView(props: MapViewProps) {
...
                 value: indicatorValue.indicatorValue ?? 0,
                 iso: indicatorValue.iso3,
+                format: indicatorValue.format,
             }),
         )
             .filter((item) => item.value > 0);
-        return countryIndicator ?? [];
+        const sortedData = [...(countryIndicator ?? [])];
+        sortedData.sort((a, b) => compareNumber(a.value, b.value, -1));
+        return sortedData;
```
Comment (tnagorra) [inline app/views/Dashboard/Overview/MapView/index.tsx]: No need to make a copy as `map` will already create a copy of the list.
We can easily do list.map().sort()

File: app/views/Dashboard/index.tsx [hunk 3]
```diff
@@ -135,6 +136,12 @@ function Dashboard() {
         COUNTRIES_AND_OUTBREAKS,
     );
 
+    useEffect(() => {
+        if (activeTab === 'country' && !filterValues) {
+            setFilterValues({ country: 'AFG' });
+        }
+    }, [activeTab, filterValues]);
+
```
Comment (tnagorra) [inline app/views/Dashboard/index.tsx:144]: Let's avoid using useEffect.

This will only set country to be "AFG" on the second render

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: app/components/MapLabel/styles.css (hunk 1 above)
Comment (subinasr) [reply app/components/MapLabel/styles.css:13]: Backup when linear gradient doesn't work.

File: app/views/Dashboard/index.tsx (hunk 3 above)
Comment (subinasr) [reply app/views/Dashboard/index.tsx:144]: Cannot be avoided.

File: app/views/Dashboard/Overview/MapView/index.tsx (hunk 2 above)
Comment (subinasr) [reply app/views/Dashboard/Overview/MapView/index.tsx]: Ok

File: app/components/MapLabel/styles.css (hunk 1 above)
Comment (subinasr) [reply app/components/MapLabel/styles.css:13]: meh

**Pull request merged**